### ISSUE TITLE
[Sticky] Automatically Escape Fixed Context

### ISSUE THREAD

**Issue by jlukic** [Semantic-Org/Semantic-UI#2774]
Sticky behaves erratically when a sidebar is shown because of the coordinate shift caused by `translate3d`.

Would have to determine whether it makes sense for sticky to automatically escape this particular case.

**Comment by lauri-elevant** [comment]
Fr the benefit of anyone experiencing this issue, I'll post the workaround here as well

	$('.following.menu').sticky({
		onStick: function() {
		    if ($(this).parents('.pushed').length) {
		        $(this).data('displaced', $(this).parent());
		        $(this).detach().appendTo(document.body).css('left', '-=260px');
		    }
		},
		onUnstick: function() {
		    if ($(this).data('displaced')) {
		        $(this).detach().appendTo($(this).data('displaced')).data('displaced', null);
		    }
		}
	});

See more details from #1745

**Comment by lauri-elevant** [comment]
Update: one problem with this approach is that if you were to include a `<button type="submit"></button>` into the menu, then without a `form` attribute on the button, it would cease to function. Hence, this approach is not entirely without side-effects. However, from a layout point of view I have not witnessed any problems so far.

**Comment by lauri-elevant** [comment]
Just a note that the workaround I posted has issues with `observeChanges:true`,  creating a feedback loop: the detach and move to body indeed counts as a mutation (depending on context).

Since there is no way to access the observer, it cannot be temporarily disabled while the workaround is being applied. So, if  `observeChanges:true` is needed, you likely need to roll your own.
